File: php/testcode/weakcipher_046.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_weakcipher_random_iv_per_msg
function weakcipher046(BenchmarkRequest $req): BenchmarkResponse {
    $key = random_bytes(32);
    $iv = random_bytes(12); // vuln-code-snippet safe-line php_weakcipher_random_iv_per_msg
    $tag = '';
    $enc = openssl_encrypt($req->param('data'), 'AES-256-GCM', $key, OPENSSL_RAW_DATA, $iv, $tag);
    $result = base64_encode($iv . $tag . $enc);
    return BenchmarkResponse::ok($result);
}
// vuln-code-snippet end php_weakcipher_random_iv_per_msg

File: ruby/testcode/weakcipher_046.rb
require_relative 'shared'
require 'openssl'
require 'securerandom'

# vuln-code-snippet start ruby_weakcipher_aes256_gcm_env_key
def encrypt_env_key_gcm(req)
  plaintext = req.body_str
  key    = [ENV.fetch('ENCRYPTION_KEY')].pack('H*')
  iv     = SecureRandom.random_bytes(12) # vuln-code-snippet safe-line ruby_weakcipher_aes256_gcm_env_key
  cipher = OpenSSL::Cipher.new('AES-256-GCM')
  cipher.encrypt
  cipher.key      = key
  cipher.iv       = iv
  cipher.auth_data = ''
  ciphertext = cipher.update(plaintext) + cipher.final
  tag        = cipher.auth_tag
  BenchmarkResponse.json({ iv: iv.unpack1('H*'), data: ciphertext.unpack1('H*'), tag: tag.unpack1('H*') })
end
# vuln-code-snippet end ruby_weakcipher_aes256_gcm_env_key
